docs/cv-upload-feature/context.md

```diff
@@ -0,0 +1,256 @@
+# CV Upload Feature - Technical Context
+
+## Overview
+This feature enables users to upload their CVs as part of Profile 4.0 foundation, helping daily.dev gather structured professional data for future job matching capabilities. The implementation includes drag-and-drop functionality, LinkedIn import guidance, and various UI states across desktop, tablet, and mobile platforms.
+
+## Feature Scope (Final Specs)
+
+### Core Functionality
+1. **Homepage Feed Banner** - Prominent CV upload prompt with drag & drop
+2. **Profile Page Integration** - CV upload section in profile settings  
+3. **LinkedIn Import Flow** - Guided process for exporting CV from LinkedIn
+4. **File Upload States** - Loading, success, and error states
+5. **Dismissal Flow** - Banner dismissal with toast feedback
+6. **Cross-platform Support** - Responsive design for desktop, tablet, mobile
+
+### User Experience Flow
+- Users see banner on homepage feed (if no CV uploaded yet)
+- Users can upload via drag-and-drop or browse files
+- LinkedIn import option available with step-by-step guidance
+- Success celebration modal after upload
+- Profile page shows uploaded CV with edit/reupload options
+- Dismissible banners show toast with profile section reference
+
+## Technical Architecture
+
+### Monorepo Structure
+- **packages/webapp** - Next.js app where CV upload will be implemented
+- **packages/shared** - Shared components, hooks, and utilities
+- **packages/extension** - Browser extension (may need CV data integration)
+
+### Technology Stack
+- **React 18.3.1** with Next.js 15.4.1
+- **TypeScript** for type safety
+- **TanStack Query v5** for server state management
+- **GraphQL** with graphql-request for API communication
+- **Tailwind CSS** with custom design system
+- **Jotai** for local state management
+
+### File Upload Implementation Requirements
+
+#### File Handling
+- **Supported formats**: PDF (primary), DOC/DOCX likely
+- **File size limits**: TBD (typical CV size constraints)
+- **Validation**: File type, size, and basic structure validation
+- **Storage**: Backend integration for secure file storage
```

Comment:
Is it supposed to do backend as well, since I don't see much planning on that. What will it use for storage.

Reply:
Nope. The scope was UI only